src/zsclient.cpp

```diff
@@ -38,6 +38,9 @@ namespace zsync2 {
         // using a set to avoid duplicate entries
         std::set<std::string> seedFiles;
         
+        std::string userSpecifiedUrl = "";
+        bool userSpecifiedUrlisEmpty = true;
```

Comment:
Why does this have to be an instance variable? It can be queried from the variable above.